### ISSUE TITLE
Improve component accessibility - aria-invalid

### ISSUE THREAD

**Issue by christophih** [primefaces/primefaces#1689]
To improve the accessibility of input components you shoul add the attribute "aria-invalid" to the components, which should be set when the component is invalid (e.g. failed validation).

https://www.w3.org/WAI/GL/wiki/Using_Aria-Invalid_to_Indicate_An_Error_Field

It is necessary to make forms with validation accessible!

Thanks!

**Comment by anm12** [comment]
I tried this with JAWS - at the moment there is no information if the input is valid or not. A failed validation cannot be found in the form, so "aria-invalid" should be added to the invalid fields.

**Comment by christophih** [comment]
Could you please solve this issue in one of the next ELITE releases?

**Comment by mertsincan** [comment]
I'll work on this issue.

**Comment by tandraschko** [comment]
@mertsincan Any news? I think it would be a great ARIA enhancement

**Comment by melloware** [comment]
I looked into this issue and it would be a great enhancement I was wondering if there was some way to intercept any ResponseWriter that sees the validation has failed an adds "aria-invalid" to it rather than try and add this to every component but i wasn't quite sure where this kind of code would go....

**Comment by tandraschko** [comment]
Its possible via a custom ResponseWriter but i think it's to magical. Maybe we can better use abstraction/inheritance in our input renderers in the future BUT as the markup is very different, not idea.

**Comment by melloware** [comment]
@tandraschko I have a fix for this.  I can put it in with just a few lines of code with my other PR that is currently pending. Now that I refactored everything to renderAccessibilityAttributes() its just adding a few lines of code like this to that method.

```java
        if (!component.isValid()) {
            writer.writeAttribute(HTML.ARIA_INVALID, "true", null);
        }
```
So the heavy lifting of that refactoring has made this fix simple and possible.  Let me know if you want to me to wait until that PR is accept and submit a new PR or add it to the current PR.
